rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+---
+feature: Create Connection
+authors:
+  - "Bohan Zhang"
+start_date: "2023/01/15"
+---
+
+# RFC: Create Connection
+
+## Motivation
+
+A connection is used to describe how to connect to an external system that users want to read data from. Once created, a connection is reusable across `CREATE SOURCE` and `CREATE SINK` statements. This RFC proposes a new `CREATE CONNECTION` statement to create a connection.
+
+## Design
+
+### Syntax
+
+```sql
+-- create connection
+CREATE CONNECTION (IF NOT EXIST) {{ connection_name }}
+  TO {{ connection_type }}
+  ( {{ field }} = {{ value }}, ...);
+
+-- create source with connection
+CREATE SOURCE {{ source_name }} ( {{ field }} = {{ value }}, ... )
+  FROM CONNECTION {{ connection_name }}
+  with ( {{ field }} = {{ value }}, ...);
+
+-- create sink with connection
+CREATE SINK {{ sink_name }} ( {{ field }} = {{ value }}, ... )
+  FROM {{ source_name }}
+  INTO CONNECTION {{ connection_name }}
+  with ( {{ field }} = {{ value }}, ...);
+```
+
+If any field is specified both in `with` clause and `CREATE CONNECTION` statement, the value in `with` clause will be used.
+
+#### AWS Private Link
+
+An AWS Private Link establishes a private connection between a VPC and a service hosted on AWS. It is a secure and scalable way to connect to AWS services from within a VPC. The following fields are required to create an AWS Private Link connection:
+
+```sql
+CREATE CONNECTION demo_connection TO AWS_PRIVATE_LINK
+  (SERVICE NAME = 'com.amazonaws.vpce.us-east-1.vpce-svc-xxxxxxxxxxxxxxxxx',
+  AVAILABILITY ZONES = array('us-east-1a', 'us-east-1b', 'us-east-1c');
+```
+
+|Field|Value|Required|Description|
+|--|--|--|--|
+|`SERVICE NAME`| varchar | yes | The name of the AWS PrivateLink service. |
+|`AVAILABILITY ZONES`| varchar[] | yes | The AWS availability zones in which the service is accessible. |
+
+> **Notice**:
+>
+> * After creating a connection, users must manually approve the connection from a Risingwave VPC in AWS console.
+> * When creating a private link to AWS MSK, users must fill in all availability zones of the MSK cluster.
+
+##### Internal table for AWS Private Link
+
+The `rw_aws_private_link` table stores the information of the AWS Private Link connection. The following fields are stored in the table:
```

Comment:
Metadata is stored in etcd with protobuf encoding, rather than relational tables.

Reply:
then implementing `show connections` should be enough